src/nomadic/summarize/main.py

```diff
@@ -0,0 +1,805 @@
+            return status
+        return f"{status};{QcStatus.DUPLICATE}"
+
+    for (_, _), data in df.query("sample_type == 'field'").groupby(
+        ["sample_id", "name"]
+    ):
+        # Select an index to keep, i.e. the best sample that should
+        # marked as duplicate
+        passing = data["status"] == "pass"
+        if passing.sum() == 1:
+            keep_idx = passing.idxmax()
+        elif passing.sum() > 1:
+            keep_idx = data[passing]["mean_cov"].idxmax()  # keep maximum coverage
+        else:  # none are passing, arbrarily keep first
+            keep_idx = data.index[0]
+
+        # NB: updating in-place
+        # must be a view, not a slice hence [,]
+        df.loc[data.index, "status"] = [
+            _update_duplicate(status, idx, keep_idx)
+            for idx, status in data["status"].items()
+        ]
+
+
+def add_quality_control_status_column(df: pd.DataFrame) -> None:
+    """
+    Add a QC status column in-place
+
+    Note:
+    - When we mark duplicates; we do it on an AMPLICON x SAMPLE level;
+    not on a per-sample level. So we could take amplicons from separate
+    samples to get the best data for that sample.
+
+    """
+    _add_qc_status_no_duplicates(df)
+    _mark_duplicates(df)
+
+
+# --------------------------------------------------------------------------------
+# Variant analysis
+#
+# --------------------------------------------------------------------------------
+
+
+def load_variant_summary_csv(
+    variants_csv: str, define_gene: bool = True
+) -> pd.DataFrame:
+    """
+    Load an clean `summary.variants.csv` data produced by `nomadic`
+
+    """
+
+    # Settings
+    NUMERIC_COLUMNS = ["dp", "wsaf"]
+    UNPHASED_GT_TO_INT = {
+        "./.": -1,
+        "0/0": 0,
+        "0/1": 1,
+        "1/1": 2,
+    }  # TODO What about multiallelic sites
+
+    # Load
+    variants_df = pd.read_csv(variants_csv)
+
+    # Reformat numeric columns to be floats
+    for c in NUMERIC_COLUMNS:
+        variants_df[c] = [float(v) if v != "." else None for v in variants_df[c]]
+
+    # Reformat unphased genotypes as integers
+    variants_df.insert(
+        variants_df.columns.tolist().index("gt") + 1,
+        "gt_int",
+        variants_df["gt"].map(UNPHASED_GT_TO_INT),
+    )
+
+    # Optionally reformat amplicon name to gene; assuming like  gene-...
+    if define_gene:
+        variants_df.insert(
+            variants_df.columns.get_loc("amplicon") + 1,
+            "gene",
+            [a.split("-")[0] for a in variants_df["amplicon"]],
+        )
+        variants_df.insert(
+            variants_df.columns.get_loc("gene") + 1,
+            "mutation",
+            [
+                f"{gene}-{aa_change}"
+                for gene, aa_change in zip(
+                    variants_df["gene"], variants_df["aa_change"]
+                )
+            ],
+        )
+
+    return variants_df
+
+
+def load_and_concat_variants(expt_dirs: list[str]) -> pd.DataFrame:
```

Comment:
This needs to have differing behaviour for versions of Nomadic where the unfiltered VCF file is stored; it would be preferable to work from that when aggregating variants.